prow/cmd/exporter/collector.go

```diff
@@ -169,3 +171,153 @@ func sanitizeLabelName(s string) string {
 func escapeString(v string) string {
 	return escapeWithDoubleQuote.Replace(v)
 }
+
+type prowJobLifecycleCollector struct {
+	lastKnownState map[types.UID]lastState
+	HistogramVec   *prometheus.HistogramVec
+}
+
+type lastState struct {
+	state    prowapi.ProwJobState
+	observed time.Time
+}
+
+func (pjc *prowJobLifecycleCollector) Update(oldJob *prowapi.ProwJob, newJob *prowapi.ProwJob) {
+	now := time.Now()
+	if lastKnownState, exists := pjc.lastKnownState[newJob.UID]; !exists {
```

Comment:
Maybe I am missing something here, but wouldn't it be sufficient to:
* if !exist: Insert unconditionally into map, return
* if newState == oldState: return
* Update map entry, create metric?

Reply:
I think it is necessary, to correctly detect transitions. I think that I could detect the `triggered` to `pending` transition a little bit more elegant.